docs/doc/code-review.md

```diff
@@ -248,80 +280,89 @@ Don't use inline style changing - `element.style`. In most cases this is a bad a
+A. Minimal requirements to meet:
+1. employ ES6 features like `const`, `let` etc. (with exclusion of ES6 `class` syntax)
+1. the code is very [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
+1. Requirements re **Constants**:
+   1. all numbers like block dimensions, initial locations are defined as constants
+   1. every number that has a semantic purpose (like those listed above) should be defined as constants; think of how your code **reads** - the closer to plain English the better
+   1. there are core constants and derived (calculated) constants
+      (e.g. `const FIELD_WIDTH = BLOCK_WIDTH * BLOCKS_NUMBER;`)
+   1. arrays of constants also qualify as constants
+      (e.g. `const INITIAL_POSITIONS = [1,2,3,4].map(rowNumber => rowNumber * BLOCK_HEIGHT);`)
+   1. const objects help organizing and structure const data even better
+      (e.g. `const PLAYER_CONF = { initialPosition: {x: 1, y: 5}, sprite: '...', ...etc... };`
+1. Requirements re **OOP**:
+   1. OO is implemented using JS prototype chain object model (**not** ES2015/ES6 `class` syntax)
+   1. properties common for some classes are generalized into a base class
+      (e.g. there is `Character` base class, which is extended by `Enemy` and `Player` classes)
+   1. class extension is implemented using `Subclass.prototype = Object.create(Superclass.prototype)`,
+      not `Subclass.prototype = new Superclass(params);`;
+      [Useful resource](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)
+   1. classes do not refer to any global variables, like global variable `player`, which is an instance of `Player` class
+      (referring to global constants and globals provided by the gaming platform like `Resources` is OK);
+      Hint: pass instance of a game object (or objects) as an argument to other game objects they need to interact with
+   1. Separation of Concerns principle is followed
+      (e.g. `update` method does only rendering and
+      doesn't **contain** any inline code to check e.g. collisions)
+1. Most common mistakes
+  1. Make sure `target = condition ? valueWhenConditionTrue : valueWhenConditionFalse` is used instead of
+     `condition ? target = valueWhenConditionTrue : target = valueWhenConditionFalse`;
+     [Conditional (ternary) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)
 
 ### OOP Exercise
 
 Relates to
 [OOP Exercise](https://github.com/kottans/frontend/blob/master/tasks/js-post-oop.md).
 
-Minimal requirements to meet:
-
-- [ ] Implement a base class to inherit from
-- [ ] Employ default parameters
-- [ ] Each species is represented with its own class
-- [ ] No need to specify species at instantiation
-- [ ] Classes for species that do not have hands by natural design
+1. Minimal requirements to meet:
+   1. Implement base classes the child classes inherit shared properties from
```

Comment:
Better wording:
`Implement base classes for child classes to inherit shared properties from`.